File: src/ast/PostfixExpression.java
package ast;

import java.io.StringWriter;

public class PostfixExpression extends ParentNode {
    private String _operator;

    public PostfixExpression(BaseNode type, String Operator) {
        super(NodeType.PostfixExpression, type);
        _operator = Operator;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("(");
        child.Print(writer);
        writer.write(")");
        writer.write(_operator);
    }
}

File: src/de/fearlesstobi/demangler/ast/PostfixExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class PostfixExpression(type: BaseNode?, private val operator: String) : ParentNode(NodeType.PostfixExpression, type) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("(")
        child!!.print(writer)
        writer.write(")")
        writer.write(operator)
    }

}